src/main/java/build/buildfarm/worker/CASFileCache.java

```diff
@@ -222,18 +226,21 @@ public FileVisitResult postVisitDirectory(Path dir, IOException exc) throws IOEx
   }
 
   /** must be called in synchronized context */
-  private void expireDirectory(Digest digest) throws IOException {
-    DirectoryEntry e = directoryStorage.remove(digest);
-    Path path = getDirectoryPath(digest);
-
-    for (Path input : e.inputs) {
+  private void purgeDirectoryFromInputs(Digest digest, Iterable<Path> inputs) {
+    for (Path input : inputs) {
       Entry fileEntry = storage.get(input);
 
       if (fileEntry != null) {
         fileEntry.containingDirectories.remove(digest);
       }
     }
-    removeDirectory(path);
+  }
+
+  /** must be called in synchronized context */
+  private void expireDirectory(Digest digest) throws IOException {
+    DirectoryEntry e = directoryStorage.remove(digest);
+    purgeDirectoryFromInputs(digest, e.inputs);
+    CASFileCache.removeDirectory(getDirectoryPath(digest));
```

Comment:
nit: why reference this by module name when it's the same module?

Reply:
I had it in my head that the linter threw a warning on this, guess it's just for !same-module